src/common/statement.cpp

```diff
@@ -18,49 +19,53 @@ namespace peloton {
 
 Statement::Statement(const std::string& statement_name,
                      const std::string& query_string)
-    : statement_name(statement_name), query_string(query_string) {}
+    : statement_name_(statement_name), query_string_(query_string) {
+  ParseQueryType(query_string_, query_type_);
+}
 
 Statement::~Statement() {}
 
-std::vector<FieldInfoType> Statement::GetTupleDescriptor() const {
-  return tuple_descriptor;
+void Statement::ParseQueryType(const std::string &query_string,
+                               std::string &query_type) {
```

Comment:
Are you passing query_type by reference here to avoid having to allocate a new variable?